nilearn/decomposition/_base.py

```diff
@@ -497,15 +510,21 @@ def fit(
         )
         self._raw_fit(data)
 
-        # Create and fit NiftiMapsMasker for transform
+        # Create and fit appropriate MapsMasker for transform
         # and inverse_transform
-        self.nifti_maps_masker_ = NiftiMapsMasker(
-            self.components_img_,
-            self.masker_.mask_img_,
-            resampling_target="maps",
-        )
+        if isinstance(self.masker_, SurfaceMasker):
+            self.surface_maps_masker_ = SurfaceMapsMasker(
+                self.components_img_, self.masker_.mask_img_
+            )
+            self.surface_maps_masker_.fit()
```

Comment:
maybe better to have a single `maps_masker_` attributes and maybe make `nifti_maps_masker_` just return `self.maps_masker_` in the volume case.

along with a deprecation warning that this public attribute will be removed in the future